### PR TITLE
feat: add --help flag

### LINKED ISSUE #249: cannot print usage information if `globs` present in config
To reproduce:

1. Run `markdownlint-cli2`: invalid usage; help text prints)
2. Run `markdownlint-cli2 --help`: invalid usage; help text prints (note: I am not sure `--help` is recognized by the CLI at all)
3. Use the [example config](https://github.com/DavidAnson/markdownlint-cli2/blob/HEAD/test/markdownlint-cli2-yaml-example/.markdownlint-cli2.yaml)
4. Run `markdownlint-cli2` - linting occurs (expected)
5. Run `markdownlint-cli2 --help` - linting occurs (**not expected**)
6. Comment out `globs` in config and repeat steps 4 and 5; you should see help text both times

Suggested fix:

`--help` should short-circuit and _always_ print usage information

### PR DESCRIPTION
**Pull request by boneskull**
This adds a `--help` flag which takes precedence over all other arguments or positionals.

Closes #249

* * *

I wasn't sure how/where you'd like to see a test for this sort of thing, so I just gave it a shot.

I changed the `else if`s to individual `if` statements because ESLint ("unicorn" plugin) started complaining that I should be using a `switch` statement.  It's not wrong, necessarily--but that's why I did what I did.

### CODE REVIEW

File: test/markdownlint-cli2-test.js [hunk 1]
```diff
@@ -611,3 +611,31 @@
...
+  return markdownlintCli2({    
+    "argv": [ "README.md", "--help" ],    
+    "logMessage": (msg) => stdouts.push(msg),
+    "logError": (msg) => t.fail(`message logged: ${msg}`)
+  })
+    .then((exitCode) => {
+      t.is(exitCode, 2);
+      t.regex(stdouts.join('\n'), /Syntax: markdownlint-cli2/u);
+    });
+})
```
[the remark this answers is not in the document]
Comment (boneskull) [reply test/markdownlint-cli2-test.js:641]: would recommend applying a code formatter as a pre-commit hook 😄

[the remark this answers is not in the document]
Comment (DavidAnson) [reply test/markdownlint-cli2-test.js:641]: This used to be handled by ESLint.

**Comment by boneskull** [comment]
@DavidAnson I've done as requested

**Comment by DavidAnson** [comment]
Thank you!

**Pull request merged**